codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/protocol/ServerProtocolTestGenerator.kt

```diff
@@ -357,7 +359,40 @@ class ServerProtocolTestGenerator(
                 }
             }
         } else {
-            rustWriter.rustTemplate("#{AssertEq}(parsed, expected);", *codegenScope)
+            val hasFloatingPointMembers = inputShape.members().any {
+                val target = model.expectShape(it.target)
+                (target is DoubleShape) || (target is FloatShape)
+            }
+
+            // TODO(https://github.com/awslabs/smithy-rs/issues/1147) Handle the case of nested floating point members.
+            if (hasFloatingPointMembers) {
+                for (member in inputShape.members()) {
+                    val memberName = codegenContext.symbolProvider.toMemberName(member)
+                    when (codegenContext.model.expectShape(member.target)) {
+                        is DoubleShape, is FloatShape -> {
+                            rustWriter.addUseImports(
+                                RuntimeType.ProtocolTestHelper(codegenContext.runtimeConfig, "FloatEquals").toSymbol()
+                            )
+                            rustWriter.rust(
+                                """
+                                assert!(parsed.$memberName.float_equals(&expected.$memberName),
+                                    "Unexpected value for `$memberName` {:?} vs. {:?}", expected.$memberName, parsed.$memberName);
+                                """
+                            )
+                        }
+                        else -> {
+                            rustWriter.rustTemplate(
+                                """
+                                    #{AssertEq}(parsed.$memberName, expected.$memberName, "Unexpected value for `$memberName`");
+                                    """,
+                                *codegenScope
+                            )
+                        }
+                    }
+                }
+            } else {
+                rustWriter.rustTemplate("#{AssertEq}(parsed, expected);", *codegenScope)
+            }
```

Comment:
should we being extracting some sort of `ProtocolTestCore`? The client has this behavior too right?

Reply:
100%. It's slightly different in that the client _always_ iterates over the members to do the equality comparison.

But yes, once I get all the protocol tests for restJson1 passing, I plan on extracting a lot of shared functionality between client and server protocol tests.